# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1468931967 14400
#      Tue Jul 19 08:39:27 2016 -0400
# Node ID 1bd872efdab508a38ca5f93895f593d757017e44
# Parent  62208397b99ebec424de45f8ce5d4742e9daa991
allow installed f77-fcn.h to use config info from Octave (bug #48559)

* mk-octave-config-h.sh: Also extract macros beginning with F77_USES_
from config.h.

* oct-conf-post.in.h (F77_USES_CRAY_CALLING_CONVENTION,
F77_USES_VISUAL_FORTRAN_CALLING_CONVENTION): Delete macros.

diff --git a/build-aux/mk-octave-config-h.sh b/build-aux/mk-octave-config-h.sh
--- a/build-aux/mk-octave-config-h.sh
+++ b/build-aux/mk-octave-config-h.sh
@@ -234,16 +234,17 @@ EOF
   fi
 else
   cat << EOF
 /* #  undef OCTAVE_INT_USE_LONG_DOUBLE */
 /* #  undef OCTAVE_ENSURE_LONG_DOUBLE_OPERATIONS_ARE_NOT_TRUNCATED */
 EOF
 fi
 
+sed -n 's/#\(\(undef\|define\) F77_USES_.*$\)/#  \1/p' $config_h_file
 sed -n 's/#\(\(undef\|define\) F77_FUNC.*$\)/#  \1/p' $config_h_file
 
 cat << EOF
 
 #endif
 
 #endif
 EOF
diff --git a/oct-conf-post.in.h b/oct-conf-post.in.h
--- a/oct-conf-post.in.h
+++ b/oct-conf-post.in.h
@@ -98,24 +98,16 @@ typedef unsigned long ino_t;
 #  define OCTAVE_HAVE_POSIX_FILESYSTEM 1
 #endif
 
 /* sigsetjmp is a macro, not a function. */
 #if defined (sigsetjmp) && defined (HAVE_SIGLONGJMP)
 #  define OCTAVE_HAVE_SIG_JUMP 1
 #endif
 
-#if defined (_UNICOS)
-#  define F77_USES_CRAY_CALLING_CONVENTION
-#endif
-
-#if 0
-#  define F77_USES_VISUAL_FORTRAN_CALLING_CONVENTION
-#endif
-
 #if defined (OCTAVE_ENABLE_64)
 #  define SIZEOF_OCTAVE_IDX_TYPE SIZEOF_INT64_T
 #else
 #  define SIZEOF_OCTAVE_IDX_TYPE SIZEOF_INT
 #endif
 
 /* To be able to use long doubles for 64-bit mixed arithmetics, we need
    them at least 80 bits wide and we need roundl declared in math.h.
